File: src/main/java/com/kamegatze/learnjvm/repository/users/UsersRepository.java
package com.kamegatze.learnjvm.repository.users;

import com.kamegatze.learnjvm.model.db.users.Users;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface UsersRepository extends JpaRepository<Users, UUID> {
    Optional<Users> findByLogin(String login);
}

File: src/main/kotlin/com/kamegatze/learnjvm/repositories/users/UsersRepository.kt
package com.kamegatze.learnjvm.repositories.users

import com.kamegatze.learnjvm.model.db.users.Users
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.PagingAndSortingRepository
import java.util.*

interface UsersRepository : CrudRepository<Users, UUID>, PagingAndSortingRepository<Users, UUID> {

    fun findByLogin(login: String): Users?

}